lib/pbench/server/api/resources/query_apis/datasets/__init__.py

```diff
@@ -130,11 +130,16 @@ def get_index(self, dataset: Dataset, root_index_name: AnyStr) -> AnyStr:
             index_map = Metadata.getvalue(dataset=dataset, key=Metadata.INDEX_MAP)
         except MetadataError as exc:
             self.logger.error("{}", str(exc))
```

Comment:
Can you also fix this to be just `self.logger.error(str(exc))`?

Reply:
I was trying to keep the PR focused on the API error strings with minimal impact on the surrounding code. We really need to have a consistent team policy on "opportunistic refactoring" ... while I'm generally in favor, I've frequently seen objections in PR reviews from others, and I've felt pressure to minimize that.

Reply:
> We really need to have a consistent team policy on "opportunistic refactoring"

I concur.  I would say that "a small number" of such refactorings in a modest-sized PR is reasonable and possibly even Good.  However, a large number should be spun off into their own PR; and, extra changes in a large PR are just more weight.

So, I guess it's a "judgement call".  🥴

Reply:
Yes; the problem is when "judgement" is variable and/or difficult to predict because it's ultimately at the whim of the reviewers and their mood at the moment they read it. We want something reasonable to inform a developer when actually *performing* the refactoring. I'm not sure that's really possible, unfortunately.

Reply:
Tracked in issue #3099.